src/games/strategy/triplea/ai/proAI/ProCombatMoveAI.java

```diff
@@ -655,8 +586,8 @@ private void removeTerritoriesWhereTransportsAreExposed(final Map<Territory, Pro
       alreadyAttackedWithUnits.addAll(attackMap.get(t).getUnits());
       alreadyAttackedWithUnits.addAll(attackMap.get(t).getAmphibAttackMap().keySet());
     }
-    for (final Territory t : moveMap.keySet()) {
-      moveMap.get(t).getMaxUnits().removeAll(alreadyAttackedWithUnits);
+    for (final Territory t : defendMap.keySet()) {
+      defendMap.get(t).getMaxUnits().removeAll(alreadyAttackedWithUnits);
```

Comment:
Perhaps this is an artifact of the migration to territory manager, for this section of code I would avoid exposing the `defendMap`, and instead allow `territoryManager` to manager it. For example, replacing this entire for loop with a function call like "territoryManager.removeAttackedWithUnits( t, alreadyAttackedWithUnits)"

The programming by side effects via the void method call IMO is a lesser evil than exposing and working with the data implementation of territoryManager.